tools/wpt/browser.py

```diff
@@ -111,30 +92,68 @@ def install(self, dest=None):
         from mozdownload import FactoryScraper
         import mozinstall
 
+        platform = {
+            "Linux": "linux",
+            "Windows": "win",
+            "Darwin": "mac"
+        }.get(uname[0])
+
+        if platform is None:
+            raise ValueError("Unable to construct a valid Firefox package name for current platform")
+
         if dest is None:
             dest = os.getcwd()
 
         filename = FactoryScraper('daily', branch='mozilla-central', destination=dest).download()
 
         try:
-            mozinstall.install(filename, dest)
+            install_dest = mozinstall.install(filename, dest)
         except mozinstall.mozinstall.InstallError as e:
-            if uname[0] == "Darwin":
+            if platform == "mac":
                 # mozinstall will fail here if nightly is already installed in the venv
                 # This only occurs on macOS because shutil.copy_tree() is called in
                 # mozinstall._install_dmg and will fail if the file already exists.
                 # copytree isn't used while installing on Windows/linux, so the same error
                 # won't be thrown if we try to rewrite there.
                 mozinstall.uninstall(dest+'/Firefox Nightly.app')
-                mozinstall.install(filename, dest)
+                install_dest = mozinstall.install(filename, dest)
             else:
                 raise
 
         os.remove(filename)
-        return find_executable("firefox", os.path.join(dest, "firefox"))
+
+        if platform == "linux":
+            bin_path = install_dest
+        elif platform == "win":
+            bin_path = os.path.join(install_dest, "core")
+        elif platform == "mac":
+            bin_path = os.path.join(install_dest, "Contents", "MacOS")
+
+        return find_executable("firefox", bin_path)
 
     def find_binary(self):
-        return find_executable("firefox")
+        platform = {
```

Comment:
I think we ended up with some duplicated code working out the binary path in the virtualenv, is that possible to factor into a function?

Reply:
Something like find_binary_path is definitely doable. We can pass the virtualenv object to it (instead of to find_binary) and have find_binary use it to ensure that we return the firefox on path if its not installed in _venv